firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryQueryCache.java

```diff
@@ -115,13 +115,13 @@ public void removeQueryData(QueryData queryData) {
    *
    * @return the number of targets removed
    */
-  int removeQueries(long upperBound, Set<Integer> activeTargetIds) {
+  int removeQueries(long upperBound, SparseArray<?> activeTargetIds) {
```

Comment:
The code in this function indicates to me that we always need to accept a SparseArray<Integer>. For any other sparse array, `activeTargetIds.get(targetId)` would return null. Can you explain to me why you chose the wildcard in this PR?

Reply:
Er, why would any other `SparseArray` return `null`? The target ids are the keys, and `SparseArray` is required to use integer keys. It doesn't matter what the value is, we are only ever using the keys. Because we don't care about the value, the `?` is appropriate. It indicates we will not make use of the value, which we don't.

Reply:
As per our offline discussion, this is ok. It's just not the most obvious use of an array-like data structure to me. I wish it was possible to make the key-only nature more obvious (such as with a Map's keySet), but since that's not possible without a copy, this LGTM